lib/install/management/upgrade.go

```diff
@@ -69,17 +69,35 @@ func (d *Dispatcher) Upgrade(vch *vm.VirtualMachine, conf *config.VirtualContain
 	if err = d.uploadImages(settings.ImageFiles); err != nil {
 		return errors.Errorf("Uploading images failed with %s. Exiting...", err)
 	}
+	var deleteImage bool
+	defer func() {
```

Comment:
emm... I don't think defer itself make it complicated. delete or not itself is not a straightforward thing. And it combines with snapshot, which should be removed as well.

To add this is because I actually forgot to delete image in some branches before. I think this will make us more careful to add bail branch.